packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -168,6 +168,35 @@ class PictureLayer extends Layer {
   }
 }
 
+/// A composited layer that maps a backend texture to a rectangle.
+///
+/// Texture layers are always leaves in the layer tree.
+class TextureLayer extends Layer {
+  /// Creates a texture layer bounded by [rect] and with backend texture
+  /// identified by [textureId].
+  TextureLayer({
+    @required this.rect,
+    @required this.textureId,
+  });
+
+  /// Bounding rectangle of this layer.
+  final Rect rect;
+
+  /// Identity of the backend texture.
+  final int textureId;
```

Comment:
what's the namespace of the `textureId`s? If a developer starts passing random values here, or walks up all the values possible, what are the implications?